system/jlib/jsocket.cpp

```diff
@@ -4801,6 +4806,13 @@ class CSocketEpollThread: public CSocketBaseThread
             if ( !si->del && (si->sock==sock) )
                 si->del = true;
         }
+        if ( !sock || !nfy )
+        {
+            WARNLOG("EPOLL: adding fd but sock or nfy is NULL");
+            selectvarschange = true;
+            triggerselect();
```

Comment:
Is there any reason this isn't at start of add() ?
And if not and nothing has actually changed don't see why selectvarschanged set to true or triggerselect().
Isn't returning false more appropriate ? And/or perhaps a dbgassertex